superset/views/utils.py

```diff
@@ -196,6 +199,26 @@ def get_form_data(  # pylint: disable=too-many-locals
     return form_data, slc
 
 
+def add_sqllab_custom_filters(form_data: Dict[Any, Any]) -> Any:
+    """
+    SQLLab can include a "filters" attribute in the templateParams.
+    The filters attribute is a list of filters to include in the
+    request. Useful for testing templates in SQLLab.
+    """
+    try:
+        data = json.loads(request.data)
+        if isinstance(data, dict):
+            params_str = data.get("templateParams")
+            if isinstance(params_str, str):
+                params = json.loads(params_str)
+                if isinstance(params, dict):
+                    filters = params.get("filters")
+                    if filters:
+                        form_data.update({"filters": filters})
+    except (TypeError, json.JSONDecodeError):
+        data = {}
```

Comment:
I wonder if we should make `filters` a reserved keyword in template params and not persist that in the virtual dataset metadata? Similarly, we probably want to make sure any other Jinja context predefined functions (`filter_values` etc) or others defined in `JINJA_CONTEXT_ADDONS` are also not overridden. Any thoughts on this?

Reply:
Sure we can make it a reserve keyword. How would that be done?

I'll need a bit of details. "Not persist that in virtual dataset metadata" ?

Reply:
I now removed `filters` from the templateParams before saving the dataset. I think this makes sense since it's really only used to test the templates.

With regards to collisions with `filter_values` this does not occur. From a templateParams if a user specifies `filter_values` it's not seen by the template. The `filter_values` remains active.